src/main/typescript/data/foundryCommon.ts

```diff
@@ -69,6 +70,26 @@ export interface FoundryCompendiumData<T> {
   flags: Record<string, unknown>;
 }
 
+export function compDataZodSchema(
+  dataSchema: z.Schema,
+  defaultImg: string,
+  defaultName: string
+): z.Schema {
+  return z
+    .object({
+      _id: z.string().default(""), // TODO: figure out how this should be handled
+      name: z.string().default(defaultName),
+      type: z
+        .enum(["character", "ammo", "apparel", "misc", "weapon", "magic"])
+        .default("character"), // TODO: find a way to DRY this up
+      data: dataSchema.default({}),
+      img: z.string().default(defaultImg),
+      effects: z.object({}).default([]), // TODO: figure out wtf to put in here
+      flags: z.array(z.object({})).default([]) // TODO: figure out wtf to put in here
```

Comment:
I think there's a record function we could use here. Would be pretty much `Record<string, unknown>`.